### PR TITLE
Release 0.4.0-rc.2

### LINKED ISSUE #253: Bug fixes to be included in rc.2
**Problem description**
The aggregated list of issues indicating needed corrections in Commonalities

- https://github.com/camaraproject/Commonalities/issues/249
  - https://github.com/camaraproject/Commonalities/pull/256
- https://github.com/camaraproject/Commonalities/issues/250
  - https://github.com/camaraproject/Commonalities/pull/251
- https://github.com/camaraproject/Commonalities/issues/252
  - https://github.com/camaraproject/Commonalities/pull/255
- https://github.com/camaraproject/Commonalities/issues/254
  - https://github.com/camaraproject/Commonalities/pull/257
- https://github.com/camaraproject/Commonalities/issues/260
  - https://github.com/camaraproject/Commonalities/pull/261
- https://github.com/camaraproject/Commonalities/issues/262
  - https://github.com/camaraproject/Commonalities/issues/271
  - https://github.com/camaraproject/Commonalities/pull/267
- https://github.com/camaraproject/Commonalities/issues/265
  - https://github.com/camaraproject/Commonalities/pull/270

**Additional context**
The time plan for RC.2 shall be agreed in the Commonalities meeting on ~~22.07.2024~~ 05.08.2024.

### PR DESCRIPTION
**Pull request by rartych**
#### What type of PR is this?
* subproject management

#### What this PR does / why we need it:
This is Release Pr for RC.2

#### Which issue(s) this PR fixes:

Fixes #253 

#### Changelog input

```
Commonalities Release 0.4.0-rc.2

```

#### Additional documentation 

https://wiki.camaraproject.org/display/CAM/Meta-release+Process#MetareleaseProcess-Milestoneactivities

### CODE REVIEW

**Comment by hdamker** [comment]
@rartych please add @camaraproject/release-management_maintainers if ready for review

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,13 +2,38 @@
...
 - [v0.3.0](#v030)
 - [v0.2.0](#v020)
 - [v0.1.0 - Initial version](#v010---initial-version)
 
 
+# v0.4.0-rc.2
+
+**This is the second Release Candidate version for Commonalities release 0.4.0.**
+
+It contains the following corrections compared to [v0.4.0-rc.1](#v040-rc1).
```
Comment (PedroDiez) [inline CHANGELOG.md:17]: Would be needed to add https://github.com/camaraproject/Commonalities/pull/267 once merged.

LGTM in advance

**Review by PedroDiez — APPROVED** [review]
LGTM

**Review by PedroDiez — APPROVED** [review]
LGTM

**Review by hdamker — APPROVED** [review]
Looks good from Release Management perspective.

- Please consider to resolve #274 and #275 before the public release
- Regarding the changes from #267 and #256 I will check across the Sub Projects and open issues as needed to make them aware.
- The PR for the public release of 0.4.0 should be created and discussed in Commonalities call on August 19th, so that the TSC on 22nd can approve it (based on the recommendation from Commonalities and Release Management).

**Review by shilpa-padgaonkar — APPROVED** [review]
/LGTM

**Comment by shilpa-padgaonkar** [comment]
As our third codowner is on vacation, merging the PR w/o his approval.

**Pull request merged**